kafka/src/test/java/org/apache/calcite/adapter/kafka/KafkaAdapterTest.java

```diff
@@ -73,6 +74,7 @@
 
   @Test void testFilterWithProject() {
     assertModel(MODEL)
+        .with(CalciteConnectionProperty.TOP_DOWN_OPT.camelName(), false)
```

Comment:
skip using top-down optimization here. This case will produce another plan with the same cost with the expect plan 
 (two candidates are both in the memo) but it will hang when executing the new plan. Still don't know why it hangs.

Reply:
@XuQianJin-Stars Do you have any insight?